### PR TITLE
FIX: Store custom emojis as uploads.

### PR DESCRIPTION
**Pull request by tgxworld**
* Depending on a hardcoded directory was a flawed design
  which made it impossible to debug when custom emojis go
  missing.

### CODE REVIEW

File: app/jobs/regular/rebake_custom_emoji_posts.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module Jobs
+  class RebakeCustomEmojiPosts < Jobs::Base
+    def execute(args)
+      name = args[:name]
+      Post.where("raw LIKE '%:#{name}:%'").each { |post| post.rebake! }
```
Comment (ZogStriP) [inline app/jobs/regular/rebake_custom_emoji_posts.rb]: I think you want `find_each` here ;)

File: app/jobs/scheduled/clean_up_uploads.rb [hunk 2]
```diff
@@ -22,11 +22,13 @@ def execute(args)
         .joins("LEFT JOIN user_avatars ua ON (ua.gravatar_upload_id = uploads.id OR ua.custom_upload_id = uploads.id)")
         .joins("LEFT JOIN user_profiles up ON up.profile_background = uploads.url OR up.card_background = uploads.url")
         .joins("LEFT JOIN categories c ON c.uploaded_logo_id = uploads.id OR c.uploaded_background_id = uploads.id")
+        .joins("LEFT JOIN custom_emojis ce ON ce.upload_id = uploads.id")
```
Comment (ZogStriP) [inline app/jobs/scheduled/clean_up_uploads.rb]: Just a note for myself: we gotta stop adding more `joins` to that query and instead add a new column to the `uploads` table in order to store where the upload is being used (or if it's being used).

File: app/models/upload.rb [hunk 3]
```diff
@@ -92,7 +97,7 @@ def self.svg_whitelist_xpath
   # options
   #   - content_type
   #   - origin (url)
-  #   - image_type ("avatar", "profile_background", "card_background")
+  #   - image_type ("avatar", "profile_background", "card_background", "custom_emoji")
```
Comment (ZogStriP) [inline app/models/upload.rb]: Nice touch updating the comment 👍

File: spec/models/custom_emoji_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe CustomEmoji, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (ZogStriP) [inline spec/models/custom_emoji_spec.rb]: What's the purpose of this?

File: app/jobs/regular/rebake_custom_emoji_posts.rb (hunk 1 above)
Comment (tgxworld) [inline app/jobs/regular/rebake_custom_emoji_posts.rb]: Good point :+1:

File: spec/models/custom_emoji_spec.rb (hunk 4 above)
Comment (tgxworld) [inline spec/models/custom_emoji_spec.rb]: Oops automatically generated by RSpec.

**Pull request merged**